### ISSUE TITLE
OnlyM crash when playing videos

### ISSUE THREAD

**Issue by AlexandrePRUVOT** [AntonyCorbett/OnlyM#91]
Hello,
We installed OnlyM in our Kingdom Hall. It is a simple and very effective tool. I really like.
And, this week ... a bug: Unable to play videos and OnlyM is blocked. I was forced to finish the OnlyM task in the Task Manager. The second screen was reserved by OnlyM …
![bugonlym](https://user-images.githubusercontent.com/44783825/48283461-dff06d00-e45c-11e8-9b8b-b67224846832.PNG)
![bugonlym2](https://user-images.githubusercontent.com/44783825/48283462-dff06d00-e45c-11e8-9542-fcf2ae5cd27c.PNG)

I found a solution: I can play videos by running OnlyM as administrator (I think @TizianoNoto will say that it is not good).
I failed to reproduce this bug on my home computer.

I think that the first video that could not be played was in 720p, 16: 9, HD, 10mn13s, 118Mb
This video could be played with VLC, Movies & TV (the Windows 10 player), Ffmpeg
Other videos could not be played with OnlyM.
I turned off and turned on the computer.
In the OnlyT settings, I tried to clear the cache, change the rendering engine and the options of "Videos and Audio", ... (one by one) but nothing worked.
I uninstalled and reinstalled OnlyT but nothing worked.
By chance, I tried to run OnlyM as an administrator and there, all the videos are played.
When I check off to run as administrator, OnlyM can not play videos.
The computer is equipped with Windows 10 Home, Intel Core I5 7400, 8GB DDR4, MSI GeForce GTx 1050 2GB graphics card. Video outputs 1) brother‘s screen: Samsung 22 '' 2) screen for timer OnlyT: Samsung 22 '' 3) Foxcon: 3 auditorium screens connected in coaxial cable 4) Small screen in back stage - Acer 19 '' ( duplicate #3)

Yesterday, I was present during the crash. When I saw my little brothers in panic, I stepped in and managed to end the OnlyM task quickly. But, I understand that there was a crash the day before or even before but I do not know how they came out (they did not think to report it to me ...)
Alexandre

**Comment by AntonyCorbett** [comment]
@AlexandrePRUVOT please would you send the log file?

**Comment by AlexandrePRUVOT** [comment]
Hello @AntonyCorbett,
Of course, log files!
Here they are.
[bugonlym.zip](https://github.com/AntonyCorbett/OnlyM/files/2568344/bugonlym.zip)
This morning, I did the following small test:
- I ran OnlyM as administrator. I played a video. And It worked.
- I unchecked "Run OnlyM Administrator". And when I played the same video, OnlyM crashed.
King regards.
Alexandre

**Comment by AntonyCorbett** [comment]
Please would you run the DriectX diagnostics and send the result? Run "dxdiag" and use the "Save All Information" button

**Comment by AntonyCorbett** [comment]
@AlexandrePRUVOT I assume that the "Media Foundation" engine works ok?

**Comment by AlexandrePRUVOT** [comment]
Hello @AntonyCorbett,
I just did some tests: OnlyM can not play videos with the "Mediafoundation" engine.
And even worse, when I run OnlyM as administrator with the "Mediafoundation" engine, OnlyM crashes when it tries to play a video.
And when I select "Ffmpeg engine" and run OnlyM as administrator, Only works again.
Here is the log file today. (I started my tests at 5:50 PM)
And here is the dxdiag file that you asked me.
Thank you again for all your efforts.
Best regard.
Alexandre
[log-20181111.txt](https://github.com/AntonyCorbett/OnlyM/files/2569756/log-20181111.txt)
[DxDiag.txt](https://github.com/AntonyCorbett/OnlyM/files/2569757/DxDiag.txt)

**Comment by AntonyCorbett** [comment]
@AlexandrePRUVOT Sorry for the delay in responding. Perhaps you would you try updating your video driver to a GeForce WHQL certified one?

**Comment by AntonyCorbett** [comment]
@AlexandrePRUVOT Also try it without butt loaded

**Comment by AlexandrePRUVOT** [comment]
Hello Antony,
I'm on the road this week and I have a professional project to close. So I'm not going to have time to change the PC settings until next Saturday. Sorry.
In addition, I want to take the time to intervene. There are two graphics cards on this PC. The motherboard and a GeForce GTX 1050. And, it is absolutely necessary that after my intervention everything works. So we'll have to do some tests.
Another project is to switch from Skype Basic to LiveKingdomHall (currently we call remote listeners. And we are not able to make the order prevail. So we will need a broadcasting software (Butt-https://danielnoethen.de/... or some other software...).
Best regards
Alexandre

**Comment by timjobutler** [comment]
@AlexandrePRUVOT Another option for broadcasting software is OBS by Streamlabs. It is free and open code.

https://github.com/stream-labs/streamlabs-obs/

**Comment by AlexandrePRUVOT** [comment]
Thanks @timjobutler,
I will try this as soon as possible.
Best regard.
Alexandre

**Comment by AlexandrePRUVOT** [comment]
Hello @AntonyCorbett,
I stopped butt. But that was not the problem.
The Intel Graphics 630 internal graphics card driver was up to date. (Driver 02/28/2018 - V23.20.16.4973).
I have updated the driver of the Nvidia GeForce GTX1050 board (Driver : from V23.21.13.8813 - 10/27/2017 to V24.21.13.9924 -05/09/2018).
At first, in the NVIDIA's settings, the sound was linked to the HDMI output (the big screens of the hall) AND OnlyM works in this case !!!!
I changed the NVIDIA settings to put the sound output back to the speakers in the hall. BUT OnlyM started crashing again! (When I select again HDMI output, it works).
The PC,has 3 sound cards and a Virtual Mixer (VoiceMeeter Banana). With this, it's possible to control all audio streams: recordings, PC sound playback, skype listening, skype answer, skype listener conversation with sound manager without interfering with the hall or recordings, ...
I checked the sound card updates. But that did not change anything.
I updated VoiceMeeter Banana. But that did not change anything. I have been using VoiceMeeter for years. And, I never had any worries with it.
The error log indicated: [Error] DirectSoundPlayer faulted.
ArgumentException: Incorrect parameter. (Exception from HRESULT: 0x80070057 (E_INVALIDARG))
When I run OnlyM as an administrator, I can still play the videos.
Best regard.
Alexandre

[log-20181118.txt](https://github.com/AntonyCorbett/OnlyM/files/2593073/log-20181118.txt)

**Comment by AntonyCorbett** [comment]
@AlexandrePRUVOT The only suggestion I can make at this stage is to reduce or turn off the sound card acceleration setting (i.e. for the sound card that is failing). If that works it would seem to indicate a bug in the driver's directX support and you may be able to find an updated driver on the manufacturer's website.

**Comment by 123jachin** [comment]
We are having a similar issue running audio files. We have been running OnlyM for almost three months now and the only issue we have had is it freezing up when playing the dramatic bible reading. Only way to close program is task manager. This has happened once during the meeting, we figured operator error, and once today preparing for the meeting. Audio files are mp3. Not a huge issue as long as the operator knows to use task manager.
Thanks Jachin 
[log-20181121.txt](https://github.com/AntonyCorbett/OnlyM/files/2605851/log-20181121.txt)

**Comment by richfrantz** [comment]
Have you considered streaming them instead?
________________________________

**Comment by 123jachin** [comment]
We might do that at some point richfrantz, at this point we are still trying to get everyone trained to JwLib, OnlyM, and OnlyR. Add in the prerecorded comments that have to be downloaded and played with vlc, its just a lot to take in.

**Comment by richfrantz** [comment]
Ok, we found streaming easier. I am assuming you're talking about the segments for the Bible study. Just click and go. The branch made it super easy.
________________________________

**Comment by 123jachin** [comment]
Correct, and it is a whole lot easier the snipping the required audio out  each week. But we just merged and until everyone gets the basics under control, we don't want to throw any curve-balls. When you look at our system it looks overwhelming. 
![img_20181121_160711](https://user-images.githubusercontent.com/45244759/48870792-ca0d7100-eda7-11e8-99c2-25ba233c4f5c.jpg)

**Comment by richfrantz** [comment]
I don't mean to sound like an idiot, but I can't even imagine the horror of snipping audio files every week. It's this easy:

Create a new desktop short cut to https://www.jw.org/en/publications/books/jesus/

Before the meeting, open it, navigate to "tonight's" lesson. Click the scripture when called for.
________________________________

**Comment by 123jachin** [comment]
It's not hard, takes 5, 10 min
